packages/nimble-components/src/mapping/base/index.ts

```diff
@@ -0,0 +1,14 @@
+import { attr } from '@microsoft/fast-element';
+import { FoundationElement } from '@microsoft/fast-foundation';
+import type { MappingKey } from './types';
+
+/**
+ * Base class for mapping configuration elements
+ */
+export abstract class Mapping extends FoundationElement {
+    @attr({ mode: 'fromView' })
+    public key?: MappingKey;
```

Comment:
In my ideal world, `MappingKey` would be provided via a generic on the type, so we could declare `MappingText` as `MappingText<string>`, so that it wouldn't be possible in TS to assign `key` on `MappingType` a value of `type number`. However, I don't think there is a means to enforce that the generic type only be a type as defined by `MappingKey` (doing `Mapping<T extends MappingKey>` opens up the type to non-supported types, which we don't want).

All of this is to simply say, I guess this is the best we can do?